pkg/archive/archive.go

```diff
@@ -225,6 +225,60 @@ func CompressStream(dest io.Writer, compression Compression) (io.WriteCloser, er
 	}
 }
 
+// ReplaceFileTarWrapper converts inputTarStream to a new tar stream
+// with a single file called header.Name with contents of file.
+// TODO: make this into a generic tar conversion function with walkFn argument
```

Comment:
Yes, it could be like a pipeline of modifications. The same pipeline steps could be also used for tar creation and extraction. Atm there is big list of specific arguments, for example user remapping, filename renaming for cp etc. while all these could be implemented as a conversion step.